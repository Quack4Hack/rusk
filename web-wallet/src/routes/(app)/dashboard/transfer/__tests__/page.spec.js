import { afterEach, describe, expect, it, vi } from "vitest";
import { cleanup, render } from "@testing-library/svelte";
import Transfer from "../+page.svelte";

vi.useFakeTimers();

describe("Transfer", () => {
  afterEach(cleanup);

  it("should render the transfer page", async () => {
    const { container } = render(Transfer);

    expect(container.firstChild).toMatchSnapshot();
  });
});
