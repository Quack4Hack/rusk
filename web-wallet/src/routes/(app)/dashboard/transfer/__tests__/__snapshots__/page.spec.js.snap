// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Transfer > should render the transfer page 1`] = `
<div
  class="dusk-card dusk-card--gap-medium"
>
  <div
    class="dusk-card__header-container"
  >
    <header
      class="dusk-card__header"
      slot="header"
    >
      <div
        class="dusk-card__heading-container dusk-card__heading-container--reverse"
      >
        <svg
          class="dusk-icon dusk-icon--size--default dusk-card__heading-icon"
          role="graphics-symbol"
          viewBox="0 0 24 24"
        >
          <path
            d="M9,3L5,7H8V14H10V7H13M16,17V10H14V17H11L15,21L19,17H16Z"
          />
        </svg>
        
         
        <h3
          class="h4"
        >
          Transfer
        </h3>
      </div>
    </header>
  </div>
   
  <div
    class="dusk-card__body-container"
  >
    <dl
      class="contract-statuses svelte-19157rx"
    >
      <dt
        class="contract-statuses__label svelte-19157rx"
      >
        Spendable
         
      </dt>
      <dd
        class="contract-statuses__value svelte-19157rx"
      >
        <span>
          0.000000000
        </span>
         
        <svg
          class="dusk-icon dusk-icon--size--default contract-statuses__icon"
          data-tooltip-id="main-tooltip"
          data-tooltip-text="DUSK"
          role="graphics-symbol"
          viewBox="0 0 24 24"
        >
          <path
            d="M12.335 0.005a12.186 12.186 0 0 0 -2.58 0.204C4.2 1.259 0 6.14 0 11.999 0 17.859 4.203 22.74 9.759 23.79c0.726 0.138 1.476 0.21 2.241 0.21 6.789 0 12.261 -5.64 11.991 -12.49C23.741 5.249 18.599 0.173 12.335 0.005Zm0.207 21.58c-0.204 0.012 -0.336 -0.222 -0.21 -0.384 1.974 -2.541 3.153 -5.733 3.153 -9.202 0 -3.468 -1.176 -6.663 -3.153 -9.202 -0.126 -0.162 0.003 -0.396 0.207 -0.384 5.052 0.282 9.06 4.465 9.06 9.586 0 5.121 -4.005 9.304 -9.057 9.586Z"
          />
        </svg>
        
         
      </dd>
    </dl>
     
    <menu
      class="contract-operations svelte-1iyylnh"
    >
      <li
        class="contract-operations__operation svelte-1iyylnh"
      >
        <button
          class="dusk-button dusk-button--type--button dusk-button--variant--primary dusk-button--size--default dusk-icon-button--labeled contract-operations__operation-button"
          type="button"
        >
          <svg
            class="dusk-icon dusk-icon--size--default dusk-button__icon"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,7L17,12H14V16H10V12H7L12,7M19,21H5A2,2 0 0,1 3,19V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5V19A2,2 0 0,1 19,21M19,19V5H5V19H19Z"
            />
          </svg>
          
           
          <span
            class="dusk-button__text"
          >
            Send
          </span>
          
        </button>
      </li>
      <li
        class="contract-operations__operation svelte-1iyylnh"
      >
        <button
          class="dusk-button dusk-button--type--button dusk-button--variant--tertiary dusk-button--size--default dusk-icon-button--labeled contract-operations__operation-button"
          type="button"
        >
          <svg
            class="dusk-icon dusk-icon--size--default dusk-button__icon"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,17L7,12H10V8H14V12H17L12,17M5,3H19A2,2 0 0,1 21,5V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V5A2,2 0 0,1 5,3M5,5V19H19V5H5Z"
            />
          </svg>
          
           
          <span
            class="dusk-button__text"
          >
            Receive
          </span>
          
        </button>
      </li>
       
      <li
        class="contract-operations__operation svelte-1iyylnh"
      >
        <a
          aria-disabled="false"
          class="dusk-anchor dusk-anchor--on-surface dusk-anchor-button dusk-anchor-button--variant--tertiary dusk-anchor-button--size--default dusk-icon-button--labeled contract-operations__operation-button"
          href="/some-base-path/dashboard"
        >
          <svg
            class="dusk-icon dusk-icon--size--default dusk-anchor-button__icon"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z"
            />
          </svg>
          
           
          <span
            class="dusk-anchor-button__text"
          >
            Back
          </span>
          
          
        </a>
      </li>
    </menu>
    
  </div>
   
</div>
`;
